sdk/core/azure-core/azure/core/pipeline/policies/_authentication.py

```diff
@@ -172,6 +252,55 @@ def on_exception(self, request):
         return
 
 
+class MultitenantCredentialPolicy(BearerTokenCredentialPolicy):
+    """Adds a bearer token Authorization header to requests, for the tenant provided in authentication challenges.
+
+    See https://docs.microsoft.com/azure/active-directory/develop/claims-challenge for documentation on AAD
+    authentication challenges.
+
+    :param credential: The credential.
+    :type credential: ~azure.core.TokenCredential
+    :param str scopes: Lets you specify the type of access needed.
+    :keyword bool enable_tenant_discovery: Determines if tenant discovery should be enabled. Defaults to True.
+    :keyword bool enable_scopes_discovery: Determines if scopes from authentication challenges should be provided to
+        token requests, instead of the scopes given to the policy's constructor. Defaults to True.
+    :raises: :class:`~azure.core.exceptions.ServiceRequestError`
+    """
+
+    def __init__(self, credential: "TokenCredential", *scopes: str, **kwargs: "Any") -> None:
+        self._discover_tenant = kwargs.pop("enable_tenant_discovery", True)
+        self._discover_scopes = kwargs.pop("enable_scopes_discovery", True)
+        super().__init__(credential, *scopes, **kwargs)
+
+    def on_challenge(self, request: "PipelineRequest", response: "PipelineResponse") -> bool:
+        """Authorize request according to an authentication challenge
+
+        This method is called when the resource provider responds 401 with a WWW-Authenticate header.
+
+        :param ~azure.core.pipeline.PipelineRequest request: the request which elicited an authentication challenge
+        :param ~azure.core.pipeline.PipelineResponse response: the resource provider's response
+        :returns: a bool indicating whether the policy should send the request
+        """
+        try:
+            challenge = _HttpChallenge(request.http_request.url, response.http_response.headers.get("WWW-Authenticate"))
+            # azure-identity credentials require an AADv2 scope but the challenge may specify an AADv1 resource
+            scope = challenge.get_scope() or challenge.get_resource() + "/.default"
+        except ValueError:
+            return False
+
+        if self._discover_tenant:
+            self.authorize_request(
+                request, scope if self._discover_scopes else self._scopes, tenant_id=challenge.tenant_id
+            )
+        else:
+            if self._discover_scopes:
```

Comment:
If looks like `discover_scopes` will only be used is `discover_tenant` is false. But they're both true by default?

Reply:
`discover_scopes` is also used on line 293 with an in-line `if` statement -- they are both true by default. They would only both be `False` if someone explicitly set them that way, which I would imagine would be the case if they wanted or needed to restrict any token requests to only use the information provided to the credential and policy. Having them both be `False` is equivalent to using the basic `BearerTokenCredentialPolicy`, so realistically, I would only see that happening if they wanted to always use this policy but toggle these options depending on something like an API version at runtime